hbase-client/src/main/java/org/apache/hadoop/hbase/filter/PrefixFilter.java

```diff
@@ -142,6 +155,31 @@ boolean areSerializedFieldsEqual(Filter o) {
     return Bytes.equals(this.getPrefix(), other.getPrefix());
   }
 
+  @Override
+  public Cell getNextCellHint(Cell cell) {
+    byte[] hintBytes;
+    if (reversed) {
+      // On reversed scan hint should be the prefix with last byte incremented
+      hintBytes = increaseLastNonMaxByte(this.prefix);
+    } else {
+      // On forward scan hint should be the prefix
+      hintBytes = prefix;
+    }
+    return PrivateCellUtil.createFirstOnRow(hintBytes, 0, (short) hintBytes.length);
+  }
+
+  private byte[] increaseLastNonMaxByte(byte[] bytes) {
+    byte[] result = Arrays.copyOf(bytes, bytes.length);
+    for (int i = bytes.length - 1; i >= 0; i--) {
+      byte b = bytes[i];
+      if (b < Byte.MAX_VALUE) {
+        result[i] = (byte) (b + 1);
+        break;
+      }
+    }
+    return result;
+  }
```

Comment:
Unfortunately I did not yet found an existing util method which would do the same.
What I tried for example `PrivateCellUtil.createFirstOnNextRow(Cell)` which is similar but not the same (and tests are failing if that is used).